src/Samples/NetCoreClient/Program.cs

```diff
@@ -1,30 +1,191 @@
-            string rawSoapResponse = ClientLogic.InvokeEchoServiceUsingWebRequest(settings.basicHttpAddress);
-            Console.WriteLine($"Http SOAP Response:\n{rawSoapResponse}");
+                switch (answer)
+                {
+                    case "1":
+                        CallBasicHttpBinding(hostAddr);
+                        break;
 
-            Console.WriteLine("Hit enter to exit");
-            Console.ReadLine();
+                    case "2":
+                        CallBasicHttpBinding_Fail(hostAddr);
+                        break;
+
+                    case "3":
+                        CallWsHttpBinding(hostAddr);
+                        break;
+
+                    case "4":
+                        CallNetTcpBinding(hostAddr);
+                        break;
+
+                    case "?":
+                        PrintScenarioList();
+                        break;
+
+                    case "": return;
+                }
+            }
+
+
+            //Settings settings = ClientLogic.BuildClientSettings(hostname);
+
+            //static void log(string value) => Console.WriteLine(value);
+            //ClientLogic.InvokeEchoServiceUsingWcf(settings, log);
+
+            //string rawSoapResponse = ClientLogic.InvokeEchoServiceUsingWebRequest(settings.basicHttpAddress);
+            //Console.WriteLine($"Http SOAP Response:\n{rawSoapResponse}");
+
+        }
+
+        private static void CallBasicHttpBinding(string hostAddr)
+        {
+            IClientChannel channel = null;
+
+            var binding = new BasicHttpBinding(IsHttps(hostAddr) ? BasicHttpSecurityMode.Transport : BasicHttpSecurityMode.None);
+            // binding.ApplyDebugTimeouts();
+
+            var factory = new ChannelFactory<IEchoService>(binding, new EndpointAddress($"{hostAddr}/EchoService/basicHttp"));
+            factory.Open();
+            try
+            {
+                IEchoService client = factory.CreateChannel();
+                channel = client as IClientChannel;
+
+                var result = client.Echo("Hello World!");
+                channel.Close();
+                Console.WriteLine(result);
+            }
+            finally
+            {
+                factory.Close();
+            }
+        }
+
+        private static void CallBasicHttpBinding_Fail(string hostAddr)
+        {
+            IClientChannel channel = null;
+
+            var binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+            // binding.ApplyDebugTimeouts();
+
+            var factory = new ChannelFactory<IEchoService>(binding, new EndpointAddress($"{hostAddr}/EchoService/basicHttp"));
+            factory.Open();
+            try
+            {
+                IEchoService client = factory.CreateChannel();
+                channel = client as IClientChannel;
+
+                var result = client.FailEcho("Hello Fault!");
+                channel.Close();
+            }
+            catch (FaultException e)
```

Comment:
@mconnew  - I am not getting a FaultException<EchoFail> here, which is what the old code was expecting. Why?